sympy/series/formal.py

```diff
@@ -1143,6 +1143,66 @@ def integrate(self, x=None, **kwargs):
 
         return self.func(f, self.x, self.x0, self.dir, (ak, self.xk, ind))
 
+    def convolve(self, other, x=None, order=4):
+        """ Convolute two Formal Power Series and return the truncated terms upto specified order.
```

Comment:
Should this return the *product* of two power series? If so, then I think that the method should be renamed.

Reply:
Yes currently it does that. But if we implement other types of convolution like `dyadic` or `cyclic` convolution, then they are no longer `product` of two power series. So I think, we should keep it `convolute`. I will try to extend the code to integrate cyclic, dyadic and other types of convolution into it.

Reply:
Hi @ArighnaIITG ,

How do you plan to add new types of convolution in the future?

If they will be separate methods, I believe renaming it to product is good like @jksuom suggested.

or one way will be to allow a ``method`` kwarg, eg. ``f1.convolve(f2, method='linear')``.